_posts/08-05-2016-measuring-your-elixir-application.md

```diff
@@ -0,0 +1,354 @@
+their required versions.
+
+After your usual `mix deps.get; mix deps.compile`, we need to configure
+`elixometer` and `exometer` OTP applications. In your `config.exs` file, add
+the following code:
+
+```elixir
+config :elixometer, reporter: :exometer_report_influxdb,
+  update_frequency: 5_000,
+  env: Mix.env,
+  metric_prefix: "myapp"
+
+config :exometer_core, report: [
+  reporters: [
+    exometer_report_influxdb: [
+      protocol: :http,
+      host: "localhost",
+      port: 8086,
+      db: "dev"
+    ]
+  ]
+]
+```
+
+With this, when starting your application you should see messages like this:
+
+```
+16:19:14.109 [info] Application lager started on node nonode@nohost
+16:19:14.196 [info] Starting reporters with [{reporters,[{exometer_report_influxdb,[{protocol,http},{host,<<"localhost">>},{port,8086},{db,<<"lu
+kla_dev">>},{tags,[{started_at,63629954320}]}]}]}]
+16:19:14.197 [info] Application exometer_core started on node nonode@nohost
+16:19:14.217 [info] Application elixometer started on node nonode@nohost
+16:19:14.290 [info] InfluxDB reporter connecting success: [{protocol,http},{host,<<"localhost">>},{port,8086},{db,<<"dev">>},{tags,[{start
+ed_at,63629954320}]}]
+16:19:14.328 [info] Running MyApp.Endpoint with Cowboy using http on port 4000
+16:19:16.976 [debug] Tzdata polling for update.
+16:19:17.006 [warning] lager_error_logger_h dropped 84 messages in the last second that exceeded the limit of 50 messages/sec
+16:19:18.569 [debug] Tzdata polling shows the loaded tz database is up to date.
+08 May 16:19:21 - info: compiled 20 files into 2 files, copied 155 in 6852ms
+```
+
+## 4th step: Visualizing the metrics
+
+Now, all we need to do is figure out how to *visualize* your metrics.
+
+[Grafana](http://grafana.org) is an open-source, general purpose dashboard and graph composer,
+which runs as a web application. It supports [Graphite](http://graphite.wikidot.com/), `InfluxDB` or [OpenTSDB](http://opentsdb.net/)
+as *backends*. `Grafana` is probably the most beautiful dashboarding solution
+out there.
+
+Setting up `Grafana` is just as easy as `InfluxDB`. We will use Docker to do
+so:
+
+```sh
+$ docker run -d -p 3000:3000 grafana/grafana:2.6.0
+```
+
+We can now log-in using the always-so-secure `admin:admin` credentials @
+<http://localhost:3000>.
+
+We now need to add our `InfluxDB` database as a data-source for grafana. To
+do it, we click at "Data Sources" and then "Add New". Fill the form like the
+picture below:
+
+![img](/public/measuring-your-elixir-application/grafanadatasource.png)
+
+(The `InfluxDB` credentials are `root:root`)
+
+With this, you're set to explore `Grafana` and create new dashboards. Below
+are some examples of our production Dashboards:
+
+![img](/public/measuring-your-elixir-application/grafanadash.png)
+
+## N'th step: Where to go from here
+
+If followed along this post, you now have a complete *time-series* storage &
+analysis suite at your disposal. You can now leverage this to create
```

Comment:
As you already referred to the reader, you can change this to be an imperative sentence.

"Leverage this tool to create indicators about yours business and make more informed decisions"

Alternatively, you can complete the paragraph with:

"Suited up executives will love to share your graphs on their shiny Prezi presentations"